Application/Features/BookingFeature/Commands/UpsertBookingCommand.cs

```diff
@@ -0,0 +1,47 @@
+﻿
+using Application.Interfaces;
+using FluentValidation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Application.Features.BookingFeature.Commands
+{
+    public abstract class UpsertBookingCommand
+    {
+        private readonly IApplicationDbContext _context;
+
+        public UpsertBookingCommand(IApplicationDbContext context)
+        {
+            _context = context;
+        }
+        public void EnsureWorkplaceIsFree(int workplaceId, DateTimeOffset startDate, DateTimeOffset endDate)
+        {
+            var busy = _context.Bookings.Where(x => 
+                          (x.WorkplaceId == workplaceId && (x.StartDate < startDate) && (startDate < x.EndDate ))
+                        ||(x.WorkplaceId == workplaceId && (x.StartDate < endDate) && (endDate < x.EndDate))
+                        ).Any();
```

Comment:
you can use Any instead of Where. + make it Async

Reply:
done